source/common/runtime/runtime_features.cc

```diff
@@ -45,6 +45,41 @@ constexpr const char* disallowed_features[] = {
     // Acts as both a test entry for deprecated.proto and a marker for the Envoy
     // deprecation scripts.
     "envoy.deprecated_features.deprecated.proto:is_deprecated_fatal",
+    // 1.12.0
+    "envoy.deprecated_features.route.proto:regex_match",
+    "envoy.deprecated_features.route.proto:regex",
+    "envoy.deprecated_features.accesslog.proto:config",
+    "envoy.deprecated_features.thrift_proxy.proto:config",
+    "envoy.deprecated_features.cds.proto:tls_context",
+    "envoy.deprecated_features.route.proto:allow_origin",
+    "envoy.deprecated_features.string.proto:regex",
+    "envoy.deprecated_features.grpc_service.proto:config",
+    "envoy.deprecated_features.tcp_proxy.proto:deprecated",
+    "envoy.deprecated_features.overload.proto:config",
+    "envoy.deprecated_features.route.proto:value",
```

Comment:
I think this is in a nested message. I'm a little worried this might accidentally conflict? Should the name actually include the full ptah?

Reply:
mmm, I agree, but I think changing naming at this point is going to be ugly.
I'll see what I can do - I think we'll have to transition over to the new naming pretty carefully.

Reply:
Oh yeah, I remember why I didn't do this - it's really non-trivial to do given the python script is fairly stateless and doesn't know the message it's in.  I think we'd have to fundamentally rewrite from scratch into something that did proto introspection.

Another alternate is we could have a not-yet-fatal section in runtime_features, and ASSERT in test that everything tagged as deprecated is either in disallowed_features or not_yet_disallowed_features.  This would CI-regression-test that folks to add their own fully qualified fields as they added code, and then we could just cut from not_yet_disallowed to disallowed as part of the release.

WDYT?

Reply:
@alyssawilk do you mind briefly syncing up with @htuch about this? Some of the proto parsing stuff he is doing now is pretty incredibly and I'm guessing it wouldn't be that difficult to change your script to actually run against the real proto tree.

Reply:
Yeah, I think the approach you could take here is to leverage the API type database to pull out this information. There may be other approaches we could take, e.g. we could include deprecation date annotations which could allow us to infer the disallowed fields via reflection.

Reply:
Sure.   Do you have someone who could take a look some time this release (at which point I can revert this) and file a tracking bug at someone else?  Alternately I could snag some time on your calendar for code pointers, given my python fu is pretty weak.

Reply:
I think anyone who can do this is probably going to be busy landing v3 for this release, so I think we can TODO/file issue here.

Reply:
OK, so I'll just land manual cleanups, and TODO script cleanups or script rewrites.